### DISCUSSION TITLE
Ruby LSPs and format on save issue

### DISCUSSION THREAD

**Discussion by tttffff** [LazyVim/LazyVim#2934] [Q&A] [answered]
I've used `{ import = "lazyvim.plugins.extras.lang.ruby" }` to get the Solargraph LSP set up. By default, it provides RuboCop diagnostics which isn't what I want. Thankfully, I can turn this reporter off in a `.solargraph.yml` file at the root of the project and it stops sending the RuboCop diagnostics to neovim. This is great, but on saving the file, it still formats the files based on those RuboCop rules.

Is there anyway to stop neovim from formatting the file based on this LSP? I also use standardrb as an LSP, and when I don't have Solargraph it formats based on standardrb which is exacly what I want... but I also want the other parts of Solargraph like goto definition etc.

Things I've tried:
- Turning the reporter off for Solargraph (outside of neovim), solves the problem of not getting diagnostics while editing.
- Telling `neovim/nvim-lspconfig` to turn formattting off for this LSP.
- Added `{ "stevearc/conform.nvim", enabled = false },` to a plugin file as a test if that is what's doing the formatting, but the issue persists.

```lua
{
  "neovim/nvim-lspconfig",
  opts = {
    servers = {
      standardrb = {},
    },
    settings = {
      solargraph = {
        autoformat = false,
        formatting = false,
      }
    }
  },
},
```

**Comment by mariochavez** [comment, marked as answer]
This is how I fixed this issue. First, I did disable `autoformat` for Solargraph, I even disabled `diagnostics` in the `nvim-lspconfig` config section.

With Mason, I installed Rubocop LSP, but not Solargraph LSP.

Then in my projects I have a `.rubocop.yml` file as this one:
```
require: standard

inherit_gem:
  standard: config/base.yml
```

Also, I have a `.standard.yml` file my configuration. Setting up the LSPs this way prevents Solargraph from formatting or reporting diagnostics, and since the Rubocop configuration is inherited from Standardrb, Rubocop LSP reports diagnostics and format code based on Standardrb rules.

**Comment by tttffff** [comment]
Thank you @mariochavez ! This does exacly what I need.

For me it was as simple as using `{ import = "lazyvim.plugins.extras.lang.ruby" }` without any manual `neovim/nvim-lspconfig` config, then creating the `.rubocop.yml` file without the `require: standard` line. I now get Solagraph working, standardrb suggestions when typing, and saving with standardrb rules. Amazing.
